### DISCUSSION TITLE
How to avoid `IconLink.internal` removing the `index.html` file?

### DISCUSSION THREAD

**Discussion by hmf** [planet42/Laika#492] [Q&A] [answered]
For the top navigation bar I am using this configuration:

```scala
    .topNavigationBar(
      // homeLink undefined, so use landing page
      navLinks = Seq(
        IconLink.internal(
          Root / "api" / "index.html",
          HeliumIcon.api,
          options = Styles("svg-link")
        ),
        IconLink.external(
          browsableLink,
          HeliumIcon.github,
          options = Styles("svg-link")
        )
      )
```

When I open the page locally I have the URL:

```
file:///mnt/ssd2/hmf/IdeaProjects/storch/out/docs/laikaLocal.dest/site/api/
```

This is correct because the root is:

```
file:///mnt/ssd2/hmf/IdeaProjects/storch/out/docs/laikaLocal.dest/site
```

I would like however to retain the `index.html` to allow for local browsing. Is this possible? Note that I am not using the [pretty urls](https://typelevel.org/Laika/latest/03-preparing-content/02-navigation.html#pretty-urls) plugin. 

TIA

**Comment by jenshalm** [comment, marked as answer]
The PrettyURLs bundle is different, it changes `foo.html` into `foo/index.html` so that no link has any `.html` extension.

For the base case where targets which are already named `index.html` have the URLs trimmed there is no configuration option to toggle this behaviour (and unfortunately it's a bit too complicated to justify a new option for this & you are also the first to ever ask for this). Which means the remaining option is to customize the renderer for `SpanLink` as described in https://typelevel.org/Laika/latest/04-customizing-laika/06-overriding-renderers.html

**Reply by hmf** [reply]
Thank you.
